### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#3]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 It seems like you want to incorporate the Wandb run name along with other metrics into the checkpoint filename. Currently, PyTorch Lightning's ModelCheckpoint may not directly support this behavior. However, probable workaround can be implemented. 

Here is a solution which makes use of a custom callback function. This function will alter the filename for the ModelCheckpoint based on the Wandb run name.

```python
import pytorch_lightning as pl
from pytorch_lightning.callbacks import ModelCheckpoint
import wandb

class CustomModelCheckpoint(ModelCheckpoint):

    def format_checkpoint_name(self, epoch: int, metrics: Dict[str, float]) -> str:
        return self.filename.format(wandb_run=wandb.run.name, epoch=epoch, val_loss=metrics["val_loss"])

class MyModel(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.model_checkpoint = CustomModelCheckpoint(monitor='val_loss', filename='{wandb_run}-{epoch}-{val_loss:.2f}')

    def configure_callbacks(self):
        return [self.model_checkpoint]
# Training code here...
```
In the above code, CustomModelCheckpoint is a subclass of ModelCheckpoint, with overridden format_checkpoint_name method. It allows you to format the filename the way you want. In the filename filed format, we have added `{wandb_run}` which gets replaced by the name of the wandb run.

Remember, the above workaround works if `wandb` is already started a run before starting the model training. Ensure that `wandb.init` is called before `trainer.fit` call.

If strictly necessary, you may raise an issue/suggestion on the PyTorch Lightning GitHub page. The maintainers can take it under consideration.
